lnrpc-ideal/rpc.proto

```diff
@@ -0,0 +1,809 @@
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+
+package lnrpc;
+
+option go_package = "github.com/lightningnetwork/lnd/lnrpc";
+
+service Lightning {
+    /** lncli: `listinvoices`
+    ListInvoices returns a list of all the invoices currently stored within the
+    database. Any active debug invoices are ignored. It has full support for
+    paginated responses, allowing users to query for specific invoices through
+    their add_index. This can be done by using either the first_index_offset or
+    last_index_offset fields included in the response as the index_offset of the
+    next request. By default, the first 100 invoices created will be returned.
+    Backwards pagination is also supported through the Reversed flag.
+    */
+    rpc ListInvoices(ListInvoiceRequest) returns (ListInvoiceResponse) {
+      option (google.api.http) = {
+        get : "/v1/invoices"
+      };
+    }
+  
+    /** lncli: `lookupinvoice`
+    LookupInvoice attempts to look up an invoice according to its payment hash.
+    The passed payment hash *must* be exactly 32 bytes, if not, an error is
+    returned.
+    */
+    rpc LookupInvoice(PaymentHash) returns (Invoice) {
+      option (google.api.http) = {
+        get : "/v1/invoice/{r_hash_str}"
+      };
+    }
+  
+    /**
+    SubscribeInvoices returns a uni-directional stream (server -> client) for
+    notifying the client of newly added/settled invoices. The caller can
+    optionally specify the add_index and/or the settle_index. If the add_index
+    is specified, then we'll first start by sending add invoice events for all
+    invoices with an add_index greater than the specified value. If the
+    settle_index is specified, the next, we'll send out all settle events for
+    invoices with a settle_index greater than the specified value. One or both
+    of these fields can be set. If no fields are set, then we'll only send out
+    the latest add/settle events.
+    */
+    rpc SubscribeInvoices(InvoiceSubscription) returns (stream Invoice) {
```

Comment:
This call doesn't return notifications for the invoice stated `Canceled` and `Accepted`.